tests/mock_tests/intfmgrd/add_ipv6_prefix_ut.cpp

```diff
@@ -0,0 +1,117 @@
+#include "gtest/gtest.h"
+#include <iostream>
+#include <fstream>  
+#include <unistd.h>
+#include <sys/stat.h>
+#include "../mock_table.h"
+#include "warm_restart.h"
+#define private public 
+#include "intfmgr.h"
+#undef private
+
+/* Override this pointer for custom behavior */
+int (*callback)(const std::string &cmd, std::string &stdout) = nullptr;
+std::vector<std::string> mockCallArgs;
+
+namespace swss {
+    int exec(const std::string &cmd, std::string &stdout)
+    {
+        mockCallArgs.push_back(cmd);
+        return callback(cmd, stdout);
+    }
```

Comment:
If my understanding is correct, this mock works due to how dynamic linker resolves symbols. Since swss::exec is defined inside tests binary it is getting used instead of swss::exec from libswsscommon.so. That means every test, not just add_ipv6_prefix_ut tests will use this version of swss::exec, however, they won't have an API to provide mock implementation. Is my undestanding correct?

Reply:
tests_intfmgrd will be compiled into a different binary so the other orchgent tests are not affected.